src/components/user/uc/users.uc.js

```diff
@@ -0,0 +1,75 @@
+const { ObjectId } = require('mongoose').Types;
+
+const { GeneralError, NotFound } = require('../../../errors');
+const { userRepo, accountRepo, trashbinRepo } = require('../repo/index');
+
+const deleteUser = async (id, app) => {
+	try {
+		// get user
+		const user = await userRepo.getUser(id, app);
+		if (!(user && user._id)) {
+			throw new NotFound(`User ${id} not found`);
+		}
+
+		const trashBin = await trashbinRepo.createUserTrashbin(id);
+		if (!(trashBin && trashBin._id)) {
+			throw new GeneralError(`Unable to initiate trashBin`);
+		}
+
+		// save user to trashbin
+		await trashbinRepo.updateUserTrashbin(trashBin._id, { user });
+
+		// get account
+		const account = await accountRepo.getUserAccount(id, app);
+
+		// save account to trashbin
+		await trashbinRepo.updateUserTrashbin(trashBin._id, { account });
+
+		// delete account
+		await accountRepo.deleteUserAccount(id);
+
+		// replace user by tombstone
+		const uid = ObjectId();
+		await userRepo.replaceUserWithTombstone(id, app, {
```

Comment:
it might seem overkill, but I would like you to move the call to the repo with the correct replacement data into a seperate function, to keep this one clear of any implementation details that are not exclusively orchestration.

Doesnt have to be as complex as the commented out version below, could simply be a function that is called with a userid, and does nothing but adding the replacement data and knowing the correct call to the repo.